pkg/controller/controller_utils.go

```diff
@@ -444,3 +447,21 @@ func InferObjectKind(obj runtime.Object) (schema.GroupVersionKind, error) {
 	}
 	return gvks[0], nil
 }
+
+// GuaranteedUpdate will retry the updateFunc to mutate the object util success, updateFunc is expected to
+// capture the object reference from the caller context to avoid unnecessary type casting.
+func GuaranteedUpdate(cli client.Client, obj runtime.Object, updateFunc func() error) error {
```

Comment:
Will this be used in the controller in the future? Seems this is used in test only.

Reply:
Yes, some get-modify-update logic could be simplified with this method